maldoca/ole/oss_utils.cc

```diff
@@ -115,27 +117,420 @@ bool BufferToUtf8::Init(const char* encode_name) {
+            {"l5", 28599},
+            {"ibm920", 28599},
+            {"8859-9", 28599},
+            {"iso-ir-148", 28599},
+            {"latin5", 28599},
+            {"csisolatin5", 28599},
+            {"iso8859-9", 28599},
+            {"iso8859_9", 28599},
+            {"ibm-920", 28599},
+            {"cp920", 28599},
+            {"8859-13", 28603},
+            {"iso-8859-13", 28603},
+            {"iso8859_13", 28603},
+            {"iso8859-13", 28603},
+            {"l9", 28605},
+            {"latin9", 28605},
+            {"csisolatin9", 28605},
+            {"iso8859_15_fdis", 28605},
+            {"ibm-923", 28605},
+            {"iso8859-15", 28605},
+            {"ibm923", 28605},
+            {"iso-8859-15", 28605},
+            {"latin0", 28605},
+            {"iso8859_15", 28605},
+            {"cp923", 28605},
+            {"x-europa", 29001},
+            {"iso-8859-8-i", 38598},
+            {"csiso2022jp", 50221},
+            {"iso-2022-jp", 50222},
+            {"jis", 50222},
+            {"iso2022-jp", 50222},
+            {"jis-encoding", 50222},
+            {"iso2022jp", 50222},
+            {"iso-2022-jp2", 50222},
+            {"csiso2022jp2", 50222},
+            {"csjisencoding", 50222},
+            {"iso2022kr", 50225},
+            {"iso2022-kr", 50225},
+            {"csiso2022kr", 50225},
+            {"iso-2022-kr", 50225},
+            {"x-cp50227", 50227},
+            {"eucjp", 51932},
+            {"euc_jp_linux", 51932},
+            {"euc-jp-linux", 51932},
+            {"euc-jp", 51932},
+            {"x-euc-jp", 51932},
+            {"euc_jp", 51932},
+            {"x-eucjp", 51932},
+            {"euc-cn", 51936},
+            {"euc_cn", 51936},
+            {"euccn", 51936},
+            {"ibm-euccn", 51936},
+            {"x-euc-cn", 51936},
+            {"ksc5601-1987", 51949},
+            {"euc_kr", 51949},
+            {"ksc5601_1987", 51949},
+            {"cp970", 51949},
+            {"ks_c_5601-1987", 51949},
+            {"ibm-euckr", 51949},
+            {"euc-kr", 51949},
+            {"euckr", 51949},
+            {"5601", 51949},
+            {"ibm-970", 51949},
+            {"ksc_5601", 51949},
+            {"hz-gb-2312", 52936},
+            {"gb18030", 54936},
+            {"gb18030-2000", 54936},
+            {"ibm-1392", 54936},
+            {"windows-54936", 54936},
+            {"x-iscii-de", 57002},
+            {"x-iscii-be", 57003},
+            {"x-iscii-ta", 57004},
+            {"x-iscii-te", 57005},
+            {"x-iscii-as", 57006},
+            {"x-iscii-or", 57007},
+            {"x-iscii-ka", 57008},
+            {"x-iscii-ma", 57009},
+            {"x-iscii-gu", 57010},
+            {"x-iscii-pa", 57011},
+            {"utf-7", 65000},
+            {"utf8", 65001},
+            {"utf_8", 65001},
+            {"utf-8", 65001}};
+    return map;
+  }();
+
+  std::string encode_name_lower(encode_name);
+  strlwr(&encode_name_lower[0]);
```

Comment:
you should use absl::AsciiStrToLower to avoid messing with underline data for the c function. In particular, I don't know if it's guaranteed encode_name_lower will necessarily be null terminated.